scripts/utilities/supported_boards.py

```diff
@@ -1,19 +1,32 @@
-def get_board_names():
+import os.path
+from os import path, listdir
+
+
+def pathIsFolderAndHasIOCFile(pathName: str):
+    isFolder = not path.isfile(pathName)
+    if not isFolder:
+        return False
+    hasIOC = any(i.endswith(".ioc") for i in listdir(pathName)) or any(
+        [i.endswith(".ioc") for i in listdir(path.join(pathName, "src/cubemx"))])
+    return hasIOC
+
+
+def get_board_names() -> tuple[str]:
     """
-    Return a tuple of board names supported in our current build
+    Return a tuple of board names supported in our current build, whose STM code can be generated (only usage)
     """
-    SUPPORTED_BOARD_NAMES = (
-        'FSM',
-        'PDM',
-        'BMS',
-        'DCM',
-        'DIM',
-        'GSM',
-    )
+    pathToBoards = path.join(path.dirname(__file__), '..', '..', 'firmware', 'thruna')
+    SUPPORTED_BOARD_NAMES: tuple[str, ...] = tuple(
```

Comment:
`SUPPORTED_BOARD_NAMES` ->`supported_board_names` (caps are supposed to be reserved for constants)